### PR TITLE
Modernize Objective-C code

### PR DESCRIPTION
**Pull request by gskbyte**
This Pull request does the following:
- Adds nullability specifiers
- Sets the type of time intervals from `int` to `NSTimeInterval`

### CODE REVIEW

File: Pod/Classes/PhoenixClient.h [hunk 1]
```diff
@@ -9,10 +9,10 @@
 #import <PhoenixClient/PhxPush.h>
 
 
-//! Project version number for PhoenixClient.
+// ! Project version number for PhoenixClient.
```
Comment (gskbyte) [inline Pod/Classes/PhoenixClient.h]: this change has to be undone, it's been created by `uncrustify`

File: Pod/Classes/PhxPush.m [hunk 2]
```diff
@@ -11,37 +11,35 @@
...
-@property (nonatomic, retain) NSDictionary *payload;
+@property (nonatomic) NSString *event;
+@property (nonatomic) NSString *refEvent;
+@property (nonatomic) NSDictionary *payload;
 
-@property (nonatomic, copy) After afterHook;
-@property (readwrite) int afterInterval;
-@property (nonatomic, retain) NSTimer *afterTimer;
+@property (nullable, nonatomic, copy) After afterHook;
+@property (nonatomic) NSTimeInterval afterInterval;
```
Comment (gskbyte) [inline Pod/Classes/PhxPush.m:24]: this property has been made `nonatomic` on purpose, because it wasn't being used in any place where it needed to be `atomic`

File: Pod/Classes/PhxChannel.h [hunk 3]
```diff
@@ -26,12 +27,12 @@
 @property (nonatomic, weak) id<PhxChannelDelegate> delegate;
 @property (nonatomic, weak) PhxSocket* socket;
 @property (nonatomic, readonly) ChannelState state;
-@property (nonatomic, retain) NSString* topic;
-@property (nonatomic, retain) NSDictionary *params;
+@property (nonatomic) NSString* topic;
+@property (nonatomic) NSDictionary *params;
```
Comment (JoachimK) [inline Pod/Classes/PhxChannel.h]: would use `strong` here, assuming the original author wanted to make the memory-management explicit here.

File: Pod/Classes/PhxPush.h [hunk 4]
```diff
@@ -9,17 +9,21 @@
...
+- (instancetype)initWithChannel:(PhxChannel*)channel
+                          event:(NSString*)event
+                        payload:(NSDictionary*)payload;
 
 - (void)send;
 
-- (PhxPush*)onReceive:(NSString*)status callback:(OnMessage)callback;
-- (PhxPush*)after:(int)ms callback:(After)callback;
+- (PhxPush *)onReceive:(NSString *)status callback:(OnMessage)callback;
+- (PhxPush *)after:(NSTimeInterval)ms callback:(After)callback;
```
Comment (JoachimK) [inline Pod/Classes/PhxPush.h]: this is not actually a correct change _if the original parameter was in milliseconds_. The parameter sounds like it's in milliseconds (`ms`), if that's the case `NSTimeInterval` is different because it's base-size is seconds. 

I'm all for using `NSTimeInterval` but if you do, you have to convert the value from milliseconds to seconds.

File: Pod/Classes/PhxPush.m [hunk 5]
```diff
@@ -51,7 +49,7 @@ - (id)initWithChannel:(PhxChannel*)channel
 }
 
 - (void)send {
-    const NSString *ref = [self.channel.socket makeRef];
+    NSString *ref = [self.channel.socket makeRef];
```
Comment (JoachimK) [inline Pod/Classes/PhxPush.m:52]: why did you remove the const here?

File: Pod/Classes/PhxPush.m [hunk 6]
```diff
@@ -76,7 +74,7 @@ - (PhxPush*)onReceive:(NSString *)status callback:(OnMessage)callback {
     return self;
 }
 
-- (PhxPush*)after:(int)ms callback:(After)callback {
+- (PhxPush*)after:(NSTimeInterval)ms callback:(After)callback {
```
Comment (JoachimK) [inline Pod/Classes/PhxPush.m]: ok, looking at the implementation it seems like the `ms` parameter was always used like a time-interval, so actually declaring it like one seems to be fine...

In that case, I think it makes sense to also rename the parameter to avoid the confusion with `milliseconds`.

File: Pod/Classes/PhxChannel.h (hunk 3 above)
Comment (gskbyte) [inline Pod/Classes/PhxChannel.h]: makes sense, but it's again a lot of changes :)

File: Pod/Classes/PhxPush.h (hunk 4 above)
Comment (gskbyte) [inline Pod/Classes/PhxPush.h]: no, if you review the usage, it's passed to `NSTimer` as-is, so what I actually should do is rename the parameter variable

File: Pod/Classes/PhxPush.m (hunk 5 above)
Comment (gskbyte) [inline Pod/Classes/PhxPush.m:52]: because it was throwing warnings and it's not needed

**Comment by gskbyte** [comment]
Thanks @JoachimK !
